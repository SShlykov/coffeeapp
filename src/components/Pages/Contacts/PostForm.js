import React, {Component} from "react";
import cDataService from "../../../services/cDataService";

export default class Form extends Component {
    state = {
      name: "",
      email: "",
      phone: "",
      message: "",
      data: {},
    };
    cDataService = new cDataService();
    onType = ( e ) => {
        let val = e.target.value;
        let attr = e.target.id;
        this.setState({[attr]:val});
    };
    updateDataObj = (name, email, phone, message)=> {
        this.setState({
            data: {name:name, email:email, phone:phone, message: message}
        });
    };
    submitFunc = async (e) =>{
        const {name, email, phone, message} = this.state;
        e.preventDefault();
        await this.updateDataObj(name, email, phone, message);
        await this.cDataService.postMessage(this.state.data);
        await this.props.submitListener();
    };
    render() {
        const  {name, message, phone, email} = this.state;
        return (
            <form onSubmit={this.submitFunc} method="POST" className="contact-form">
                <label htmlFor="name" className="col-6 contact-label importantInput">Name</label>
                <input
                    type="name"
                    id="name"
                    className="col-6 contact-input"
                    value={name}
                    onChange={this.onType}
                />
                <label htmlFor="e-mail" className="col-6 contact-label importantInput">E-mail</label>
                <input
                    type="e-mail"
                    id="email"
                    className="col-6 contact-input"
                    value={email}
                    onChange={this.onType}
                />
                <label htmlFor="phone" className="col-6 contact-label">Phone</label>
                <input
                    type="phone"
                    id="phone"
                    placeholder="+7(...)...-...."
                    className="col-6 contact-input"
                    value={phone}
                    onChange={this.onType}
                />
                <label htmlFor="message" id="labelMes" className="col-12 text-center contact-label importantInput">Your
                    message</label>
                <textarea
                    placeholder="Tell us..."
                    name="message"
                    id="message"
                    className="col-12"
                    value={message}
                    onChange={this.onType}
                />
                <div className="centred">
                    <input
                        type="submit"
                        value="Send us"
                        className="anotherButton"
                    />
                </div>
            </form>
        )
    }
};
