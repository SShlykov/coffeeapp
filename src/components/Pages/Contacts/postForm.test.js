import React from 'react';
import PostForm from "./PostForm";
import {shallow} from 'enzyme';

describe("Testing <PostForm />", () => {
    const form = shallow(<PostForm/>);
    describe("Testing snap & state", ()=>{
        it("PostForm has rendered correctly", () => {
            expect(form).toMatchSnapshot();
        });
        it("PostForm state 'name' is empty string" , ()=>{
            expect(form.state().name).toBe("");
        })
        it("PostForm state 'email' is empty string" , ()=>{
            expect(form.state().email).toBe("");
        })
        it("PostForm state 'phone' is empty string" , ()=>{
            expect(form.state().phone).toBe("");
        })
        it("PostForm state 'message' is empty string" , ()=>{
            expect(form.state().message).toBe("");
        })
        it("PostForm state 'data' is empty object" , ()=>{
            expect(form.state().data).toBeObject();
        })
        it("PostForm state 'errorName' is empty string" , ()=>{
            expect(form.state().errorName).toBe("");
        })
        it("PostForm state 'errorEmail' is empty string" , ()=>{
            expect(form.state().errorEmail).toBe("");
        })
        it("PostForm state 'errorEmail' is empty string" , ()=>{
            expect(form.state().errorEmail).toBe("");
        })
        it("PostForm state 'errorMessage' is empty string" , ()=>{
            expect(form.state().errorMessage).toBe("");
        })
    })
    describe("Handlers tests", () => {
        it("testing updateDataObj", () => {
            let name = "sergey", 
                email = "sshlykov@icovt.ru", 
                phone = "+79130024328",
                message = "Hello World"
            form.instance().updateDataObj(name, email, phone, message);
            expect(form.state().data.name).toBe(name);
            expect(form.state().data.email).toBe(email);
            expect(form.state().data.phone).toBe(phone);
            expect(form.state().data.message).toBe(message);
        })
    })
})