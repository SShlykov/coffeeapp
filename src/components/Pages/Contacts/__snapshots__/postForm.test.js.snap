// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PostForm /> Testing snap & state PostForm has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="col-6 contact-label importantInput"
          htmlFor="name"
        >
          Name
        </label>,
        <input
          className="col-6 contact-input"
          id="name"
          onChange={[Function]}
          required={true}
          type="name"
          value=""
        />,
        null,
        <label
          className="col-6 contact-label importantInput"
          htmlFor="e-mail"
        >
          E-mail
        </label>,
        <input
          className="col-6 contact-input"
          id="email"
          onChange={[Function]}
          required={true}
          type="e-mail"
          value=""
        />,
        null,
        <label
          className="col-6 contact-label"
          htmlFor="phone"
        >
          Phone
        </label>,
        <InputElement
          className="col-6 contact-input"
          id="phone"
          mask="+7 (999) 999-9999"
          onChange={[Function]}
          placeholder="+7 (...) ...-...."
          type="phone"
          value=""
        />,
        <label
          className="col-12 text-center contact-label importantInput"
          htmlFor="message"
          id="labelMes"
        >
          Your message
        </label>,
        <textarea
          className="col-12"
          id="message"
          name="message"
          onChange={[Function]}
          placeholder="Tell us..."
          required={true}
          value=""
        />,
        null,
        <div
          className="centred"
        >
          <input
            className="anotherButton"
            type="submit"
            value="Send us"
          />
        </div>,
      ],
      "className": "contact-form",
      "method": "POST",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
          "className": "col-6 contact-label importantInput",
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-6 contact-input",
          "id": "name",
          "onChange": [Function],
          "required": true,
          "type": "name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-mail",
          "className": "col-6 contact-label importantInput",
          "htmlFor": "e-mail",
        },
        "ref": null,
        "rendered": "E-mail",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-6 contact-input",
          "id": "email",
          "onChange": [Function],
          "required": true,
          "type": "e-mail",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Phone",
          "className": "col-6 contact-label",
          "htmlFor": "phone",
        },
        "ref": null,
        "rendered": "Phone",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "col-6 contact-input",
          "id": "phone",
          "mask": "+7 (999) 999-9999",
          "onChange": [Function],
          "placeholder": "+7 (...) ...-....",
          "type": "phone",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your message",
          "className": "col-12 text-center contact-label importantInput",
          "htmlFor": "message",
          "id": "labelMes",
        },
        "ref": null,
        "rendered": "Your message",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-12",
          "id": "message",
          "name": "message",
          "onChange": [Function],
          "placeholder": "Tell us...",
          "required": true,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="anotherButton"
            type="submit"
            value="Send us"
          />,
          "className": "centred",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "anotherButton",
            "type": "submit",
            "value": "Send us",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="col-6 contact-label importantInput"
            htmlFor="name"
          >
            Name
          </label>,
          <input
            className="col-6 contact-input"
            id="name"
            onChange={[Function]}
            required={true}
            type="name"
            value=""
          />,
          null,
          <label
            className="col-6 contact-label importantInput"
            htmlFor="e-mail"
          >
            E-mail
          </label>,
          <input
            className="col-6 contact-input"
            id="email"
            onChange={[Function]}
            required={true}
            type="e-mail"
            value=""
          />,
          null,
          <label
            className="col-6 contact-label"
            htmlFor="phone"
          >
            Phone
          </label>,
          <InputElement
            className="col-6 contact-input"
            id="phone"
            mask="+7 (999) 999-9999"
            onChange={[Function]}
            placeholder="+7 (...) ...-...."
            type="phone"
            value=""
          />,
          <label
            className="col-12 text-center contact-label importantInput"
            htmlFor="message"
            id="labelMes"
          >
            Your message
          </label>,
          <textarea
            className="col-12"
            id="message"
            name="message"
            onChange={[Function]}
            placeholder="Tell us..."
            required={true}
            value=""
          />,
          null,
          <div
            className="centred"
          >
            <input
              className="anotherButton"
              type="submit"
              value="Send us"
            />
          </div>,
        ],
        "className": "contact-form",
        "method": "POST",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
            "className": "col-6 contact-label importantInput",
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": "Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-6 contact-input",
            "id": "name",
            "onChange": [Function],
            "required": true,
            "type": "name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-mail",
            "className": "col-6 contact-label importantInput",
            "htmlFor": "e-mail",
          },
          "ref": null,
          "rendered": "E-mail",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-6 contact-input",
            "id": "email",
            "onChange": [Function],
            "required": true,
            "type": "e-mail",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Phone",
            "className": "col-6 contact-label",
            "htmlFor": "phone",
          },
          "ref": null,
          "rendered": "Phone",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "col-6 contact-input",
            "id": "phone",
            "mask": "+7 (999) 999-9999",
            "onChange": [Function],
            "placeholder": "+7 (...) ...-....",
            "type": "phone",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your message",
            "className": "col-12 text-center contact-label importantInput",
            "htmlFor": "message",
            "id": "labelMes",
          },
          "ref": null,
          "rendered": "Your message",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-12",
            "id": "message",
            "name": "message",
            "onChange": [Function],
            "placeholder": "Tell us...",
            "required": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="anotherButton"
              type="submit"
              value="Send us"
            />,
            "className": "centred",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "anotherButton",
              "type": "submit",
              "value": "Send us",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
